var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -423,43 +209,36 @@ def cmake_args(self):
             args.append("-DWarpX_openpmd_internal=OFF")
 
         if "+python" in spec:
-            if spec.satisfies("@24.08:"):
-                args.append("-DWarpX_pyamrex_internal=OFF")
-                args.append("-DWarpX_pybind11_internal=OFF")
-                args.append(self.define_from_variant("WarpX_PYTHON_IPO", "python_ipo"))
+            pip_args = PythonPipBuilder.std_args(self) + [f"--prefix={self.prefix}"]
```

Comment:
I pushed the equivalent already for `py-amrex`:

Fix errors passing `pip` arguments using `cmake` to pip:
- In CMake, argument lists are semicolon-delimited.
- Contrary to a shell, CMake preserves spaces and even escapes them automatically using backslashes.
  - When passing spaces between options, this `cmake` passes `pip` a long option that has spaces in it!
- That means when you set a CMake variable that contains spaces, they are passed as a stringle, long string. They are not split by spaces!
  - CMake are no does not do splitting by spaces, but by semilcolon instead.
- Thus, the only way to pass a list of multiple argument to a called command like `pip` is to pass them to CMake as a CMake list that is semicolon-separated:

After the 1st commit by me, this is the 2nd fix. If there are no other issues like that, this may fix GitLab.

I'll approve when this applied and a manual build passes.
- I would like to suggest leaving the status as Draft until the reviewers signalled that they would approve the removed version and the related changes. And I suggest setting it to ready when the PR is ready for a the final GitLab CI pipeline to create the condition for the merge.

Reply:
Thanks. I arrived at the same conclusion yesterday.

This needs a mainline fix and patch first, because I do not treat them as lists yet. Will push more once this works.